### PR TITLE
port of cmake c++ standard detection from RC_1_1

### PR DESCRIPTION
**Pull request by aldenml**
This fix my environment, when `${CMAKE_CXX_STANDARD_DEFAULT} EQUAL 98` is `true`

### CODE REVIEW

**Comment by arvidn** [comment]
`CMAKE_CXX_STANDARD_DEFAULT` is what the compiler uses if nothing is specified?
Is this relevant because we assume this is the version clients will use? How do clients specify which version they use?

**Comment by arvidn** [comment]
ping @zeule

**Comment by aldenml** [comment]
@zeule I wasn't aware of that code in `RC_1_1`, I can do the port to `master`

**Comment by aldenml** [comment]
@arvidn I rebase from `master`, @zeule let me know what do you think of this change

File: cmake/Modules/LibtorrentMacros.cmake [hunk 1]
```diff
@@ -54,20 +54,35 @@ macro(find_public_dependency _name)
...
 			set(_std ${CMAKE_CXX_STANDARD_DEFAULT})
 		else()
 			set(_std ${_minimal_supported_version})
 		endif()
 	else()
-		if (${_std} LESS ${_minimal_supported_version})
-			message(FATAL_ERROR "Sorry, C++${_std} is not supported by libtorrent")
+		_cxx_standard_to_year(std_year ${_std})
+		if (${std_year} LESS ${minimal_supported_version_year})
+			message(SEND_ERROR "Sorry, C++${_std} is not supported by libtorrent with the given Boost version")
```
Comment (zeule) [inline cmake/Modules/LibtorrentMacros.cmake]: AFAIR, the master does not connect boost version and required C++ standard, hence this message would be misleading.

Comment (aldenml) [reply cmake/Modules/LibtorrentMacros.cmake]: true, removing it

**Pull request merged**